modules/GenericDataParser/src/server/generic_data_parser.R

```diff
@@ -223,6 +223,77 @@ validateMetaData <- function(metaData, configList, username, formatSettings = li
 
   return(list(validatedMetaData=validatedMetaData, duplicateExperimentNamesAllowed=duplicateExperimentNamesAllowed, useExisting=useExisting))
 }
+
+
+saveIncomingEndpointData <- function(codes, codeKind) {
+  # Saves incoming column data into their codetable / data dictionaries
+  # set up for endpoint manager data, only looking at "column name", "column units", "concentration units", "time units" for now... 
+  # We are going off just starting to just take one input columnName to save
+  # 
+  # Args:
+  #   codes:      array of the names of the endpoint values
+  #   codeKind:   codeKind that this is for (valid options: column name, column units, concentration units, time units)
+  #
+  # Returns:
+  #   N/A
+
+  library(dplyr)
+
+  # Remove NAs from the codes
+  codes <- codes[!is.na(codes)]
+
+  # If there are no codes after removing NAs, end the script early
+  if (length(codes) == 0) {
+    return (NULL)
+  }
+
+  # we don't want to work with other codeKinds (for endpoint manager only right now... )
+  validCodeKinds = c("column name", "column units", "concentration units", "time units")
+
+  #Before we save the incoming code, we need to check if the code/codeKind pair already exists... 
+  #get relevant ddict data
+  ddictLists = getDDictValuesByTypeKindFormat(lsKind = codeKind, lsType = "data column")
+  ddictValues <- rbindlist(lapply(ddictLists, jsonlite::fromJSON))
+
+  # if the object is empty, we assume that no codes exist yet, so we will save all the input codes
+  if(nrow(ddictValues) == 0) {
+    codesToSave <- codes
+
+  #if the object is not empty, we will check to see which of the existing codes overlap with our input codes
+  } else {
+    #extract all the codes for a given codeKind
+    codesForCodeKind = ddictValues %>% filter(lsKind == codeKind) %>% pull(codeName)
```

Comment:
I'm not sure `codeName` is the right attribute to pull here. If I'm understanding the flow so far,
codeName will be something like 'DDICT-00001" whereas the `shortName` attribute will be the relevant string containing the column name or units. Could you double check this?

You'll also be able to test this by uploading the same column name in two experiments, and seeing if you get duplicate entries in the Column Name dropdown. You can also confirm in the `ddict_value` table

Reply:
You are completely right! I think I misread the table. 

<img width="530" alt="Screen Shot 2022-11-16 at 12 58 39 PM" src="https://user-images.githubusercontent.com/107148842/202257392-d7fb168f-be77-4c83-8678-d9708a89cc3b.png">